app/helpers/hosts_helper.rb

```diff
@@ -72,44 +72,45 @@ def last_report_tooltip(record)
   end
 
   # method that reformat the hostname column by adding the status icons
-  def name_column(record)
-    label = record.host_status
-    case label
-    when "Pending Installation"
-      style ="label-info"
-      # TRANSLATORS: host's status: first character of "build"
-      short = s_("Build|B")
-    when "Alerts disabled"
-      style = "label-default"
-      # TRANSLATORS: host's status: first character of "disabled"
-      short = s_("Disabled|D")
-    when "No reports"
-      style = "label-default"
-      # TRANSLATORS: host's status: first character of "no reports"
-      short = s_("No reports|N")
-    when "Out of sync"
-      style = "label-warning"
-      # TRANSLATORS: host's status: first character of "sync" (out of sync)
-      short = s_("Sync|S")
-    when "Error"
-      style = "label-danger"
-      # TRANSLATORS: host's status: first character of "error"
-      short = s_("Error|E")
-    when "Active"
-      style = "label-info"
-      # TRANSLATORS: host's status: first character of "active"
-      short = s_("Active|A")
-    when "Pending"
-      style = "label-warning"
-      # TRANSLATORS: host's status: first character of "pending"
-      short = s_("Pending|P")
-    else
-      style = "label-success"
-      # TRANSLATORS: host's status: first character of "OK"
-      short = s_("OK|O")
+  def name_column(host)
+    style = host_global_status_icon_class_for_host(host)
+    tooltip = host.host_statuses.map { |status| "#{_(status.name)}: #{_(status.to_label)}" }.join(', ')
```

Comment:
`host.host_statuses` may need some explicit ordering for consistency (here and elsewhere), since there's no default scope ordering.